src/lib/OpenEXRCore/openexr_conf.h

```diff
@@ -0,0 +1,46 @@
+/*
+** SPDX-License-Identifier: BSD-3-Clause
+** Copyright Contributors to the OpenEXR Project.
+*/
+
+#ifndef OPENEXR_CONF_H
+#define OPENEXR_CONF_H
+#pragma once
+
+#include <OpenEXRConfig.h>
+
+/*  TODO: Move this stuff to OpenEXRConfig.h.in? */
+
+/** 
+ * @defgroup configurable name wrappers for C
+ * @brief These are a group of macros which enable people to define a custom name
+ *
+ * This might be used to put this version of the C layer into a custom
+ * "namespace" so as not to conflict with a system library used by an
+ * existing DCC or similar.
+ *
+ * @{
+ */
+
+/** Used to declare / use types */
+#define EXR_TYPE(n) exr_ ## n
+/** Used to declare / use enums */
+#define EXR_DEF(n) EXR_ ## n
+/** Used to declare / call functions */
+#define EXR_FUN(n) exr_ ## n
+
```

Comment:
Would it be helpful to have

    #define EXR_TYPE(n) exr_ ## n ## _t

to more easily distinguish types from functions and be sure they won't clash?

I'm a little leery of these macros. Maybe it's nothing to worry about, but it makes me immediately wonder if they will throw off modern editor's (such as Sublime, VSCode, CLion, etc.) ability to correctly syntax highlight, refactor, connect declarations and usage of functions, show information about a function's declarations and parameter when you hover over a call site, etc.

Reply:
sure, the _t is a good point, not sure why I didn't do that, I obviously subconsciously was thinking it with the separation.

But these macros are the biggest discussion I'd like to have - now is the time to decide whether to use it or not. And I think in libraries like jpeg where you can have 8-bit and 12-bit variants present, it makes sense. I was initially thinking people might like to customize for their facility, but I don't think this is that big a deal at this level, because the file format is the file format, and we should do the symbol versioning thing to manage different versions.

Reply:
I'm curious about Larry's point on the IDEs ~ do the common IDEs recognize the expanded function names such that "go to definition/declaration" works? As a sanity check, GLEW's macros tend to work in IDEs, so it might be interesting to see if an editor can eat GLEW, can it eat these macros as well?